components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultServiceURLBuilder.java

```diff
@@ -124,7 +124,8 @@ protected String getResolvedUrlPath(String tenantDomain) {
         if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled() && !resolvedUrlContext.startsWith("t/") &&
                 !resolvedUrlContext.startsWith("o/")) {
             if (mandateTenantedPath || isNotSuperTenant(tenantDomain)) {
-                if (PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId() != null) {
+                if (PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId() != null &&
```

Comment:
Shall we add a comment as well

Reply:
Also, don't we have to change line 129 as follows?
`resolvedUrlStringBuilder.append("/o/").append(PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId());`

` resolvedUrlStringBuilder.append("/o/").append(tenantDomain);` - works fine for the moment since tenant domain == org id except the super tenant

Reply:
A Comment is added.
Yes, better to get the organization id.

Fixed by - https://github.com/wso2/carbon-identity-framework/pull/4246/commits/dd11bcae1710e222e4985deff7a3ecf69c05333f